code/parachain/frame/pablo/src/lib.rs

```diff
@@ -1331,6 +1337,9 @@ pub mod pallet {
 		}
 
 		#[transactional]
+		// FIXME(benluelo): This function gets the pool from storage, then calls
```

Comment:
Does this actually cause any issue?

Reply:
It causes 2 extra database reads, which will increase the weight of the call significantly